scripts/core/ui/components/video.tsx

```diff
@@ -19,12 +20,23 @@ export class VideoComponent extends React.PureComponent<IProps> {
         }
 
         const videoRenditions = Object.values(item.renditions)
-            .filter((rendition) => rendition.mimetype != null && rendition.mimetype.startsWith('video'));
+            .filter(
+                (rend) => rend.mimetype != null && (rend.mimetype.startsWith('video') ||
+                                                    rend.mimetype.startsWith('application/x-mpegurl')),
+            );
+
+        if (videoRenditions.length > 0 && videoRenditions[0].mimetype === 'application/x-mpegurl') {
```

Comment:
> videoRenditions[0].mimetype === 'application/x-mpegurl'

Why are you checking only the first one? Why not map over all renditions and return the player depending on the mime type?

```
renditions.map((rendition) => {
    if (rendition.mimetype === 'application/x-mpegurl') {
        return (
            <HLSVideoComponent ... />
        );
    } else {
        return (
            <video ... />;
        );
    }
})
```

Reply:
The logic of how `HLSVideoComponent` works is different from the default `video`, there is no `source`(s) and it can handle only one stream URL at a period of time. Most probably API will return only one rendition with `application/x-mpegurl` and the rest are images.  For example, Talpa's API returns it like that.

Reply:
@tomaskikutis later I'll create a fake PR to trigger CI to build a test instance, so you can test it with a proper backend.

Reply:
@tomaskikutis actually, I think this part must be improved, theoretically item's renditions might contain mp4 and m3u8 and in this case, we will show whether `default` player or `HLSVideoComponent ` depends on index order which is wrong.

Reply:
> it can handle only one stream URL at a period of time

I see. That's an important detail. How about handling it in the code so it's clear? You can filter out the renditions so it includes only one `'application/x-mpegurl'` and proceed with mapping it as I suggested previously.

```
let hlsVideoCount = 0;

renditions
    .filter((rendition) => {
        if (rendition.mimetype === 'application/x-mpegurl') {
            hlsVideoCount++;
            
            if (hlsVideoCount === 1) {
                return true;
            } else {
                logger.error(new Error('HLSVideoComponent can handle only one stream URL at a period of time.'));
                return false;
            }
        } else {
            true;
        }
    })
    .map((rendition) => {
        if (rendition.mimetype === 'application/x-mpegurl') {
            return (
                <HLSVideoComponent ... />
            );
        } else {
            return (
                <video ... />;
            );
        }
    })
```